From 2b00515a9b639fd1e057f3ebf10ded2dde920764 Mon Sep 17 00:00:00 2001
From: Christian Franke <chris@opensourcerouting.org>
Date: Mon, 30 Sep 2013 12:27:49 +0000
Subject: [PATCH] bgpd, ospfd, zebra: fix some DEFUN definitions

Fixup some DEFUNS with incorrect command strings or mixed up helpstrings.

Signed-off-by: Christian Franke <chris@opensourcerouting.org>
Signed-off-by: David Lamparter <equinox@opensourcerouting.org>
---
 bgpd/bgp_route.c  | 58 +++++++++++++++++++++++++++----------------------------
 ospfd/ospf_vty.c  |  2 ++
 zebra/debug.c     |  6 +++---
 zebra/rtadv.c     |  2 +-
 zebra/zebra_vty.c |  8 ++++----
 5 files changed, 39 insertions(+), 37 deletions(-)

diff --git a/bgpd/bgp_route.c b/bgpd/bgp_route.c
index fb35fab..335543e 100644
--- a/bgpd/bgp_route.c
+++ b/bgpd/bgp_route.c
@@ -4375,7 +4375,7 @@ ALIAS_DEPRECATED (bgp_network_mask_natural,
        "AS-Pathlimit TTL, in number of AS-Path hops\n")
 ALIAS_DEPRECATED (bgp_network_mask_natural_backdoor,
        bgp_network_mask_natural_backdoor_ttl_cmd,
-       "network A.B.C.D backdoor pathlimit (1-255>",
+       "network A.B.C.D backdoor pathlimit <1-255>",
        "Specify a network to announce via BGP\n"
        "Network number\n"
        "Specify a BGP backdoor route\n"
@@ -6796,7 +6796,7 @@ DEFUN (show_ip_bgp_view,
        IP_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n")
+       "View name\n")
 {
   struct bgp *bgp;
 
@@ -6818,7 +6818,7 @@ DEFUN (show_ip_bgp_view_route,
        IP_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Network in the BGP routing table to display\n")
 {
   return bgp_show_route (vty, argv[0], argv[1], AFI_IP, SAFI_UNICAST, NULL, 0);
@@ -6831,7 +6831,7 @@ DEFUN (show_ip_bgp_view_prefix,
        IP_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "IP prefix <network>/<length>, e.g., 35.0.0.0/8\n")
 {
   return bgp_show_route (vty, argv[0], argv[1], AFI_IP, SAFI_UNICAST, NULL, 1);
@@ -7904,14 +7904,14 @@ DEFUN (show_bgp_view_afi_safi_community_all,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
 #ifdef HAVE_IPV6
        "Address family\n"
 #endif
        "Address Family modifier\n"
        "Address Family modifier\n"
-       "Display routes containing communities\n")
+       "Display routes matching the communities\n")
 {
   int afi;
   int safi;
@@ -7945,7 +7945,7 @@ DEFUN (show_bgp_view_afi_safi_community,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
 #ifdef HAVE_IPV6
        "Address family\n"
@@ -7982,7 +7982,7 @@ ALIAS (show_bgp_view_afi_safi_community,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
 #ifdef HAVE_IPV6
        "Address family\n"
@@ -8009,7 +8009,7 @@ ALIAS (show_bgp_view_afi_safi_community,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
 #ifdef HAVE_IPV6
        "Address family\n"
@@ -8040,7 +8040,7 @@ ALIAS (show_bgp_view_afi_safi_community,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
 #ifdef HAVE_IPV6
        "Address family\n"
@@ -10198,7 +10198,7 @@ DEFUN (show_bgp_view_afi_safi_neighbor_adv_recd_routes,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
 #ifdef HAVE_IPV6
        "Address family\n"
@@ -10613,7 +10613,7 @@ DEFUN (show_ip_bgp_view_rsclient,
        IP_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Information about Route Server Client\n"
        NEIGHBOR_ADDR_STR)
 {
@@ -10663,7 +10663,7 @@ DEFUN (show_bgp_view_ipv4_safi_rsclient,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Address Family modifier\n"
        "Address Family modifier\n"
@@ -10723,7 +10723,7 @@ DEFUN (show_ip_bgp_view_rsclient_route,
        IP_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Information about Route Server Client\n"
        NEIGHBOR_ADDR_STR
        "Network in the BGP routing table to display\n")
@@ -10795,7 +10795,7 @@ DEFUN (show_bgp_view_ipv4_safi_rsclient_route,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Address Family modifier\n"
        "Address Family modifier\n"
@@ -10877,7 +10877,7 @@ DEFUN (show_ip_bgp_view_rsclient_prefix,
        IP_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Information about Route Server Client\n"
        NEIGHBOR_ADDR_STR
        "IP prefix <network>/<length>, e.g., 35.0.0.0/8\n")
@@ -10949,7 +10949,7 @@ DEFUN (show_bgp_view_ipv4_safi_rsclient_prefix,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Address Family modifier\n"
        "Address Family modifier\n"
@@ -11031,7 +11031,7 @@ DEFUN (show_bgp_view_neighbor_routes,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Detailed information on TCP and BGP neighbor connections\n"
        "Neighbor to display information about\n"
        "Neighbor to display information about\n"
@@ -11057,7 +11057,7 @@ ALIAS (show_bgp_view_neighbor_routes,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Detailed information on TCP and BGP neighbor connections\n"
        "Neighbor to display information about\n"
@@ -11070,7 +11070,7 @@ DEFUN (show_bgp_view_neighbor_damp,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Detailed information on TCP and BGP neighbor connections\n"
        "Neighbor to display information about\n"
        "Neighbor to display information about\n"
@@ -11096,7 +11096,7 @@ ALIAS (show_bgp_view_neighbor_damp,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Detailed information on TCP and BGP neighbor connections\n"
        "Neighbor to display information about\n"
@@ -11109,7 +11109,7 @@ DEFUN (show_bgp_view_neighbor_flap,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Detailed information on TCP and BGP neighbor connections\n"
        "Neighbor to display information about\n"
        "Neighbor to display information about\n"
@@ -11135,7 +11135,7 @@ ALIAS (show_bgp_view_neighbor_flap,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Detailed information on TCP and BGP neighbor connections\n"
        "Neighbor to display information about\n"
@@ -11246,7 +11246,7 @@ DEFUN (show_bgp_view_rsclient,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Information about Route Server Client\n"
        NEIGHBOR_ADDR_STR)
 {
@@ -11295,7 +11295,7 @@ DEFUN (show_bgp_view_ipv6_safi_rsclient,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Address Family modifier\n"
        "Address Family modifier\n"
@@ -11354,7 +11354,7 @@ DEFUN (show_bgp_view_rsclient_route,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Information about Route Server Client\n"
        NEIGHBOR_ADDR_STR
        "Network in the BGP routing table to display\n")
@@ -11425,7 +11425,7 @@ DEFUN (show_bgp_view_ipv6_safi_rsclient_route,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Address Family modifier\n"
        "Address Family modifier\n"
@@ -11506,7 +11506,7 @@ DEFUN (show_bgp_view_rsclient_prefix,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Information about Route Server Client\n"
        NEIGHBOR_ADDR_STR
        "IPv6 prefix <network>/<length>, e.g., 3ffe::/16\n")
@@ -11577,7 +11577,7 @@ DEFUN (show_bgp_view_ipv6_safi_rsclient_prefix,
        SHOW_STR
        BGP_STR
        "BGP view\n"
-       "BGP view name\n"
+       "View name\n"
        "Address family\n"
        "Address Family modifier\n"
        "Address Family modifier\n"
diff --git a/ospfd/ospf_vty.c b/ospfd/ospf_vty.c
index 2ba8188..5e5a0b0 100644
--- a/ospfd/ospf_vty.c
+++ b/ospfd/ospf_vty.c
@@ -1976,6 +1976,8 @@ DEFUN (ospf_area_authentication_message_digest,
        ospf_area_authentication_message_digest_cmd,
        "area (A.B.C.D|<0-4294967295>) authentication message-digest",
        "OSPF area parameters\n"
+       "OSPF area ID in IP address format\n"
+       "OSPF area ID as a decimal value\n"
        "Enable authentication\n"
        "Use message-digest authentication\n")
 {
diff --git a/zebra/debug.c b/zebra/debug.c
index 7bfdb77..c3b00e0 100644
--- a/zebra/debug.c
+++ b/zebra/debug.c
@@ -35,8 +35,8 @@ DEFUN (show_debugging_zebra,
        show_debugging_zebra_cmd,
        "show debugging zebra",
        SHOW_STR
-       "Zebra configuration\n"
-       "Debugging information\n")
+       "Debugging information\n"
+       "Zebra configuration\n")
 {
   vty_out (vty, "Zebra debugging status:%s", VTY_NEWLINE);
 
@@ -128,7 +128,7 @@ DEFUN (debug_zebra_packet_detail,
        "Debug option set for zebra packet\n"
        "Debug option set for receive packet\n"
        "Debug option set for send packet\n"
-       "Debug option set detaied information\n")
+       "Debug option set detailed information\n")
 {
   zebra_debug_packet = ZEBRA_DEBUG_PACKET;
   if (strncmp ("send", argv[0], strlen (argv[0])) == 0)
diff --git a/zebra/rtadv.c b/zebra/rtadv.c
index ae5c5a1..91d7b32 100644
--- a/zebra/rtadv.c
+++ b/zebra/rtadv.c
@@ -1483,7 +1483,7 @@ DEFUN (no_ipv6_nd_router_preference,
 
 ALIAS (no_ipv6_nd_router_preference,
        no_ipv6_nd_router_preference_val_cmd,
-       "no ipv6 nd router-preference (high|medium|low",
+       "no ipv6 nd router-preference (high|medium|low)",
        NO_STR
        "Interface IPv6 config commands\n"
        "Neighbor discovery\n"
diff --git a/zebra/zebra_vty.c b/zebra/zebra_vty.c
index 45928e9..f4946f4 100644
--- a/zebra/zebra_vty.c
+++ b/zebra/zebra_vty.c
@@ -280,9 +280,9 @@ DEFUN (ip_route_mask_flags_distance,
        "IP destination prefix mask\n"
        "IP gateway address\n"
        "IP gateway interface name\n"
-       "Distance value for this route\n"
        "Emit an ICMP unreachable when matched\n"
-       "Silently discard pkts when matched\n")
+       "Silently discard pkts when matched\n"
+       "Distance value for this route\n")
 {
   return zebra_static_ipv4 (vty, 1, argv[0], argv[1], argv[2], argv[3], argv[4]);
 }
@@ -294,9 +294,9 @@ DEFUN (ip_route_mask_flags_distance2,
        "Establish static routes\n"
        "IP destination prefix\n"
        "IP destination prefix mask\n"
-       "Distance value for this route\n"
        "Emit an ICMP unreachable when matched\n"
-       "Silently discard pkts when matched\n")
+       "Silently discard pkts when matched\n"
+       "Distance value for this route\n")
 {
   return zebra_static_ipv4 (vty, 1, argv[0], argv[1], NULL, argv[2], argv[3]);
 }
-- 
1.9.2

